runners/flink/1.11/src/test/java/org/apache/beam/runners/flink/RemoteMiniClusterImpl.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.flink;
+
+import org.apache.flink.runtime.minicluster.MiniCluster;
+import org.apache.flink.runtime.minicluster.MiniClusterConfiguration;
+
+/** A {@link MiniCluster} which allows remote connections for the end-to-end test. */
+public class RemoteMiniClusterImpl extends RemoteMiniCluster {
+
+  public RemoteMiniClusterImpl(MiniClusterConfiguration miniClusterConfiguration) {
+    super(miniClusterConfiguration);
+  }
+
+  @Override
+  public int getClusterPort() {
+    return getClusterInformation().getBlobServerPort();
```

Comment:
Is it really the blob server port here or do we want the Job server RPC port?

Reply:
Yeah I really monkeyed that one :see_no_evil:

If I understood it correctly, it's a workaround for this, to expose port in `useSingleRpcService` mode:
https://github.com/apache/flink/blob/release-1.10.1/flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniCluster.java#L270

Which was changed to `createLocalRpcService` in 1.11.
https://github.com/apache/flink/blob/release-1.11.1/flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniCluster.java#L271

Pushed a fix for this.